### ISSUE TITLE
macOS file system scan affects privilege escalation areas (excluded)

### ISSUE THREAD

**Issue by erik-bershel** [anchore/syft#1367]
**Please provide a set of steps on how to reproduce the issue**
1) Install SYFT with using of default installation method on any of macOS's used in GitHub Actions
2) Run command like syft dir:/ --exclude ./Users --exclude ./System/Volumes --exclude ./private
3) Wait until one of the symlinks guide tool to the area with non-default privileges

**What happened**:
Security preferences will ask about permissions for Terminal like here:
<img width="1253" alt="Снимок экрана 2022-11-28 в 16 58 01" src="https://user-images.githubusercontent.com/110455084/204325249-842c3c4f-d3c3-41d2-ac78-fbdd7a0b2a16.png">

**What you expected to happen**:
Indexing and cataloging process ends without additional interactions

**Anything else we need to know?**:
Next strings in logs after confirmation
<img width="1254" alt="Снимок экрана 2022-11-28 в 16 58 15" src="https://user-images.githubusercontent.com/110455084/204325674-821472c6-359f-4019-81bd-3989db3f6ecd.png">

**Environment**:
- Output of `syft version`: 0.62.1, darwin, amd64
- OS (e.g: `cat /etc/os-release` or similar): macOS 12.6.1, macOS 11.7.1

**Comment by kzantow** [comment]
Hi @erik-bershel -- I think this is expected. If you scan the entire filesystem, it will attempt to access every directory, which on macOS might prompt for permissions.

Is the issue you are expecting these to be excluded but they are not because they are symlinked?

**Comment by erik-bershel** [comment]
@kzantow yep. I was expecting exceptions to avoid elevation requests that can't be tracked/resolved when using workflow. To some extent, this helped, since initially the number of such requests was much larger. But this last request (on screenshot), unfortunately, is not defeated by exceptions.

**Comment by wagoodman** [comment]
@erik-bershel when running `syft dir:/ --exclude ./Users --exclude ./System/Volumes --exclude ./private` what directory did you run this from? Was it from `/` or somewhere else? I could see if it wasn't from `/` then the `--exclude` feature would not work as expected.

@kzantow I recall that `--exclude` cannot take an absolute path, but cannot remember the specific reasons. Maybe we could reconsider this (in which case `pwd` does not factor into these kinds of issues anymore).
